### ISSUE TITLE
No time library for freeBSD?

### ISSUE THREAD

**Issue by trevorwhitney** [fsouza/fake-gcs-server#1739]
We just recently tried to update from `v1.7.0` to `v1.47.7`(I know, huge jump), and it looks like in that jump the time functions were split out to OS specific version (linux, mac, and windows). Unfortunately, this breaks our [freebsd/amd64 build](https://github.com/grafana/loki/actions/runs/10744853958/job/29802951861) complaining of:

```
vendor/github.com/fsouza/fake-gcs-server/internal/backend/fs.go:120:82: undefined: createTimeFromFileInfo
vendor/github.com/fsouza/fake-gcs-server/internal/backend/fs.go:156:82: undefined: createTimeFromFileInfo
```

Is this a known issue? Should it/can it use the linux version? Is support for freeBSD intentionally being dropped? Please let me know if I can help so we can upgrade our very out of date version. Thanks!

**Comment by fsouza** [comment]
I assume this was by accident. 

We can fix it and add a CI job that cross-compiles to freebsd. I don't have access to infrastructure to run tests on freebsd, but we can guarantee that compilation won't break again in the future at least.

**Comment by fsouza** [comment]
This is fixed and the fix was included in 1.50.0. Let me know if you still have any issues!

**Comment by trevorwhitney** [comment]
thank you, will do!
